src/Data/Lens/Index.purs

```diff
@@ -15,6 +15,33 @@ import Data.Set as S
 import Data.StrMap as SM
 import Data.Traversable (traverse)
 
+-- | Use an `Index` optic on types where: 
+-- | 1. The focus element might not be present.
+-- | 2. You either cannot or do not want to add new elements or delete existing ones. 
+-- |
+-- | `Array` is a typical example:
+-- |
+-- | ```purescript 
+-- | preview (ix 1) [0, 1, 2] == Just 1
+-- |
+-- | set (ix 1) 8888 [0, 1, 2] == [0,8888,2]
+-- | ```
+-- |
+-- | Note the use of `preview` rather `view`.
```

Comment:
Perhaps explaining why `preview` rather than `view` is worth a mention here? (because it's a `Traversal` rather than a `Lens`).